tests/pos/gadts2.scala

```diff
@@ -8,9 +8,9 @@ object Test {
   case class Cell[a](var x: a) extends Term[a]
   final case class NumTerm(val n: Number) extends Term[Number]
 
-  def f[a](t: Term[a], c: Cell[a]): Unit = {
+  def f[A](t: Term[A], c: Cell[A]): Unit = {
     t match {
-      case NumTerm(n) => c.x = MyDouble(1.0)
+      case NumTerm(n) => c.x = MyDouble(1.0)  // problem is: this assignment is not type correct, since gadt variable is forgotten
```

Comment:
I see the example compiles -- but based on `asInstanceOf` hack.

In thee code example I gave, I think the typer should reject the code -- a typing error, instead of unreachable clause in exhaustivity check.